source/ranges.tex

```diff
@@ -6096,6 +6145,303 @@
+for (span<char const> word : split(str, ' ')) {
+  for (char ch : word)
+    cout << ch;
+  cout << '*';
+}
+// The above prints: the*quick*brown*fox*
+\end{codeblock}
+\end{example}
+
+\rSec3[range.split.view]{Class template \tcode{split_view}}
+
+\begin{codeblock}
+namespace std::ranges {
+  template<@\libconcept{forward_range}@ V, @\libconcept{forward_range}@ Pattern>
+    requires @\libconcept{view}@<V> && @\libconcept{view}@<Pattern> &&
+             @\libconcept{indirectly_comparable}@<iterator_t<V>, iterator_t<Pattern>, ranges::equal_to>
+  class split_view : public view_interface<split_view<V, Pattern>> {
+  private:
+    V @\exposid{base_}@ = V();                              // \expos
+    Pattern @\exposid{pattern_}@ = Pattern();               // \expos
+    // \ref{range.split.iterator}, class \tcode{split_view::\exposid{iterator}}
+    struct @\exposid{iterator}@;                            // \expos
+    // \ref{range.split.sentinel}, class \tcode{split_view::\exposid{sentinel}}
+    struct @\exposid{sentinel}@;                            // \expos
+  public:
+    split_view() = default;
+    constexpr split_view(V base, Pattern pattern);
+
+    template<@\libconcept{forward_range}@ R>
+      requires @\libconcept{constructible_from}@<V, views::all_t<R>> &&
+               @\libconcept{constructible_from}@<Pattern, single_view<range_value_t<R>>>
+    constexpr split_view(R&& r, range_value_t<R> e);
+
+    constexpr V base() const& requires @\libconcept{copyable}@<V> { return @\exposid{base_}@; }
+    constexpr V base() && { return std::move(@\exposid{base_}@); }
+
+    constexpr @\exposid{iterator}@ begin();
+
+    constexpr auto end() {
+      if constexpr (@\libconcept{common_range}@<V>) {
+        return @\exposid{iterator}@{*this, ranges::end(@\exposid{base_}@), {}};
+      } else {
+        return @\exposid{sentinel}@{*this};
+      }
+    }
+
+    constexpr subrange<iterator_t<V>> @\exposid{find-next}@(iterator_t<V>); // \expos
+  };
+
+  template<class R, class P>
+    split_view(R&&, P&&) -> split_view<views::all_t<R>, views::all_t<P>>;
+
+  template<@\libconcept{forward_range}@ R>
+    split_view(R&&, range_value_t<R>)
+      -> split_view<views::all_t<R>, single_view<range_value_t<R>>>;
+}
+\end{codeblock}
+
+\begin{itemdecl}
+constexpr split_view(V base, Pattern pattern);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{base_} with \tcode{std::move(base)}, and
+\exposid{pattern_} with \tcode{std::move(pattern)}.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<@\libconcept{forward_range}@ R>
+  requires @\libconcept{constructible_from}@<V, views::all_t<R>> &&
+           @\libconcept{constructible_from}@<Pattern, single_view<range_value_t<R>>>
+constexpr split_view(R&& r, range_value_t<R> e);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{base_} with \tcode{views::all(std::forward<R>(r))}, and
+\exposid{pattern_} with \tcode{single_view\{std::move(e)\}}.
```

Comment:
This should have been `views::single(std::move(e))` (P2367R0 is fixing this in the previous `split_view`) - but might be a bit of a stretch to consider it editorial...

Reply:
Thanks, should I make the change as part of that paper instead then?

Reply:
Works for me - as long as @jwakely is happy about the change being applied to both splits?

Reply:
Yes, I think that's the right thing to do.

Reply:
OK, I'll incorporate that into the next motion, thank you!